Temporal Nexus: Main Encyclopedia (#3053)
temporalio/documentation: docs/encyclopedia/temporal-nexus-api-registry.mdx

Author: Separate pages are better from a SEO standpoint.

---
id: nexus-api-registry
title: Nexus API Registry
sidebar_label: Nexus API Registry

> description: Learn how to use the Nexus API Registry to manage Nexus Endpoints. 

Reviewer: See updated content, added a little bit more. Screenshots in the original doc.

toc_max_heading_level: 4
slug: /temporalnexus/api-registry
keywords:
  - API registry
  - service documentation
tags:
  - distributed systems
  - integration

>   - temporal

Author: good callout. changed the keywords/tags and kept it simple for now.

---

> ### Nexus API Registry {#nexus-api-registry}

Reviewer: incorporated this into my suggestions

> The Nexus API Registry is an account-scoped list of Nexus `Endpoints` available within an account.

Reviewer: ```suggestion
The Nexus API Registry stores information about Nexus Endpoints. In the open source version of Temporal, the registry is scoped to a cluster, while in Temporal Cloud, it is scoped to an account. Developers can advertise available endpoints and services, so others can find them for use in their caller Workflows.

Endpoint names must be unique within the registry. When the Temporal Service dispatches a Nexus request, it resolves the request's Endpoint to a Target Namespace and Task Queue via the Nexus API Registry.
```

> The Nexus API Registry is an account-scoped list of Nexus `Endpoints` available within an account.

Reviewer: updated in my suggestions

> Endpoints descriptions are rendered as markdown, and can include Service and Operation descriptions, and input/output types, so consumers know how to use the Services and Endpoint offers.

Reviewer 2: ```suggestion
Endpoints descriptions are rendered as markdown, and can include Service and Operation descriptions, and input/output types, so consumers know how to use the Services that the Endpoint offers.
```

Guessing at this meaning, not sure if it's correct

Author: I'm going to keep 'and' for now.

> Endpoints descriptions are rendered as markdown, and can include Service and Operation descriptions, and input/output types, so consumers know how to use the Services and Endpoint offers.

Reviewer 3: ```suggestion
Endpoints descriptions are rendered as Markdown, and can include Service and Operation descriptions, and input/output types, so consumers know how to use the Services and Endpoint offers.
```

> Endpoints descriptions are rendered as markdown, and can include Service and Operation descriptions, and input/output types, so consumers know how to use the Services and Endpoint offers.

Reviewer: ```suggestion
Endpoints descriptions are rendered as markdown, and can include Service and Operation descriptions, and input/output types, so consumers know how to use the Services and Endpoint offers.

You can search the Nexus API registry for Endpoint name or an endpoint Target Namespace to quickly find an Endpoint of interest.

Creating a Nexus endpoint includes setting an Access Policy of caller namespaces that can access the Endpoint. Even the Target Namespace must be added to the Access Policy to access the Endpoint. 

Editing a Nexus Endpoint allows changing everything but the Endpoint Name. The Target Namespace and Target Task Queue can be updated without interrupting existing in-flight Nexus Operations that are already being processed, and new Nexus Operations will be routed to the updated Target Namespace and Target Task Queue.

The Access Policy may also be edited to add or remove caller Namespaces that can access the Endpoint.
```

> Endpoints descriptions are rendered as markdown, and can include Service and Operation descriptions, and input/output types, so consumers know how to use the Services and Endpoint offers.

Author: moved it
